components/eamxx/src/physics/mam/eamxx_mam_microphysics_process_interface.cpp

```diff
@@ -695,6 +858,35 @@ void MAMMicrophysics::run_impl(const double dt) {
   linoz_output[5] = linoz_dPmL_dT;
   linoz_output[6] = linoz_dPmL_dO3col;
   linoz_output[7] = linoz_cariolle_pscs;
+#else
+  //FIXME: I will need to modify perform_atmospheric_chemistry_and_microphysics in MAM4xx.
+  // Let's do that after we have tested the new interface for DataInterpolation.
+  // const auto linoz_o3_clim = get_field_in("o3_clim").get_view<const Real **>();
+  // const auto linoz_o3col_clim = get_field_in("o3col_clim").get_view<const Real **>();
+  // const auto linoz_t_clim = get_field_in("t_clim").get_view<const Real **>();
+  // const auto linoz_PmL_clim = get_field_in("PmL_clim").get_view<const Real **>();
+  // const auto linoz_dPmL_dO3 = get_field_in("dPmL_dO3").get_view<const Real **>();
+  // const auto linoz_dPmL_dT = get_field_in("dPmL_dT").get_view<const Real **>();
+  // const auto linoz_dPmL_dO3col = get_field_in("dPmL_dO3col").get_view<const Real **>();
+  // const auto linoz_cariolle_pscs = get_field_in("cariolle_pscs").get_view<const Real **>();
+  m_data_interpolation->run(end_of_step_ts());
+  m_data_interpolation_linoz->run(end_of_step_ts());
+
+  const auto linoz_o3_clim = get_field_out("o3_clim").get_view<Real **>();
+  const auto linoz_o3col_clim = get_field_out("o3col_clim").get_view<Real **>();
+  const auto linoz_t_clim = get_field_out("t_clim").get_view<Real **>();
+  const auto linoz_PmL_clim = get_field_out("PmL_clim").get_view<Real **>();
+  const auto linoz_dPmL_dO3 = get_field_out("dPmL_dO3").get_view<Real **>();
+  const auto linoz_dPmL_dT = get_field_out("dPmL_dT").get_view<Real **>();
+  const auto linoz_dPmL_dO3col = get_field_out("dPmL_dO3col").get_view<Real **>();
+  const auto linoz_cariolle_pscs = get_field_out("cariolle_pscs").get_view<Real **>();
+
+  const auto oxid_O3 = get_field_out("O3").get_view<Real **>();
+  const auto oxid_OH = get_field_out("OH").get_view<Real **>();
+  const auto oxid_NO3 = get_field_out("NO3").get_view<Real **>();
+  const auto oxid_HO2 = get_field_out("HO2").get_view<Real **>();
```

Comment:
We already have all these names (linoz and oxids) in a vector. Can we use a for-loop instead of explicitly using names?